### PR TITLE
API Update listview / treeview to use pjax instead of deferred loading

### LINKED ISSUE #1645: Pages cms list view doesn't work anymore
Clicking the "list" icon does nothing.

PRs:
 - [ ] https://github.com/silverstripe/silverstripe-cms/pull/1665
 - [ ] https://github.com/silverstripe/silverstripe-framework/pull/6224

### PR DESCRIPTION
**Pull request by tractorcow**
Fixes #1645

Requires minor framework PR at https://github.com/silverstripe/silverstripe-framework/pull/6224

### CODE REVIEW

File: templates/SilverStripe/CMS/Controllers/Includes/CMSMain_PageList_Sidebar.ss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<div class="panel panel--padded cms-content-view cms-tree-view-sidebar" data-pjax-fragment="Content-PageList-Sidebar">
+	<% include SilverStripe\\CMS\\Controllers\\CMSMain_TreeView %>
```
Comment (tractorcow) [inline templates/SilverStripe/CMS/Controllers/Includes/CMSMain_PageList_Sidebar.ss]: Currently only one view allowed for sidebar; See https://github.com/silverstripe/silverstripe-cms/issues/1664 for suggestion to extend this to allow listview.

**Comment by tractorcow** [comment]
I've fixed the search issue.

**Comment by flamerohr** [comment]
Fix looks good, I reckon this is ready for merging :D

**Comment by tractorcow** [comment]
Approved so I'll merge it. :)

**Pull request merged**